lib/transfer.c

```diff
@@ -307,6 +307,12 @@ static CURLcode sendrecv_dl(struct Curl_easy *data,
                                 is_multiplex, &result);
     if(nread < 0) {
       if(CURLE_AGAIN == result) {
+        if(data->req.download_done && data->req.no_body &&
+           !data->req.resp_trailer) {
+          DEBUGF(infof(data, "EAGAIN, download done, no trailer announced, "
+                 "not waiting for EOS"));
+          data->req.keepon &= ~KEEP_RECV;
```

Comment:
Should this condition also mark the connection for close? Or can we expect us to handle later HTTP/2 frames arriving over it for this transfer to be correctly just ignored?

Reply:
Good catch. I think I'll modify that a little.

Reply:
Changed the implementation to use the same code path for a simulated as for a real EOS.